exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/spill/SpillSet.java

```diff
@@ -413,7 +419,12 @@ public SpillSet(DrillConfig config, FragmentHandle handle, PhysicalOperator popC
       fileManager = new HadoopFileManager(spillFs);
     }
 
-    spillDirName = String.format("%s_%s_%s-%s_minor%s", QueryIdHelper.getQueryId(handle.getQueryId()),
+    // If provided with a prefix to identify the Drillbit, prepend that to the
+    // spill directory.
+
+    spillDirName = String.format("%s%s_%s_%s-%s-%s",
+        ep == null ? "" : ep.getAddress() + "-" + ep.getUserPort() + "/",
```

Comment:
Any chance the address would be "" ?  Then the directory name would start with a "-" which could cause some trouble (probably only for access from the shell, thus OK).